File: languages/c++/patterns/20 FlyWeight/MoneyFactory.cpp
#include <iostream>
#include <array>
#include "Money.h"
#include "MoneyFactory.h"
#include "MetallicMoney.h"
#include "PaperMoney.h"

int MoneyFactory::objectsCount = 2;

MoneyFactory::MoneyFactory(){
	objects[0] = new MetallicMoney;
	objects[1] = new PaperMoney;
}


Money* MoneyFactory::GetMoneyToDisplay(MoneyType form) // Same as GetFlyWeight()
   {
	switch(form){
		case MoneyType::Metallic:{
			return objects[0];
		}
		case MoneyType::Paper:{
			return objects[1];
		}
		default: return nullptr;
	}	
   }

File: languages/java/20 FlyWeight/Main/src/main/MoneyFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main;

/**
 *
 * @author lupin
 */
public class MoneyFactory {
    Money[] m = new Money[]{new MetallicMoney(), new PaperMoney()};
    public MoneyFactory(){
    
    }
    public Money getMoneyToDisplay(MoneyType type){
        switch(type){
            case Metallic:{
                return m[0];
            }
            case Paper:{
                return m[1];
            }
        }
        return null;
    }
}
